### DISCUSSION TITLE
TESTING ISSUE

### DISCUSSION THREAD

**Discussion by dodeyeikoi** [Cyfrin/foundry-full-course-cu#1145] [Q&A]
I have an issue with testing out a function in the Raffle lesson.

When I call forge test on this test function below, it passes ✅ but when I add an API to it and test with sepolia, it fails ❌

how can I ensure it passes with or without the API added

````
function testPerformUpkeepRevertsIfCheckUpkeepIsFalse() public {
        // Arrange
        uint256 currentBalance = 0;
        uint256 numPlayers = 0;
        Raffle.RaffleState rState = raffle.getRaffleState();
        // Act / Assert
        vm.expectRevert(
            abi.encodeWithSelector(
                Raffle.Raffle__upKeepNotNeeded.selector,
                currentBalance,
                numPlayers,
                rState
            )
        );
        raffle.performUpkeep("");
    }
    ````
    
this is the error I get when i use ```forge test --mt testPerformUpkeepRevertsIfCheckUpkeepIsFalse --fork-url $SEPOLIA_RPC_URL````  : 

````
cryptoace@Dodeye:~/solidity-course/foundry-f23/foundry-smart-contract-lottery-f23$ forge test --mt testPerformUpkeepRevertsIfCheckUpkeepIsFalse --fork-url $SEPOLIA_RPC_URL
[⠆] Compiling...
No files changed, compilation skipped

Running 1 test for test/unit/RaffleTest.t.sol:RaffleTest
[FAIL. Reason: Error != expected error: 0x43df4e3d000000000000000000000000000000000000000000000000008e1bc9bf04000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 != 0x43df4e3d000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000] testPerformUpkeepRevertsIfCheckUpkeepIsFalse() (gas: 20115)
Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 7.73s
 
Ran 1 test suites: 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/unit/RaffleTest.t.sol:RaffleTest
[FAIL. Reason: Error != expected error: 0x43df4e3d000000000000000000000000000000000000000000000000008e1bc9bf04000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 != 0x43df4e3d000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000] testPerformUpkeepRevertsIfCheckUpkeepIsFalse() (gas: 20115)

Encountered a total of 1 failing tests, 0 tests succeeded
````

I hope I get answers please 🙏🙏 thank you

**Comment by 0xusmanf** [comment]
The problem is you are getting different values in your error msg than you are expecting them to be, why it is happening on Sepolia and not locally is probably due to the fact that you are not using your own key to deploy the contract, hence Foundry is using its default address to deploy the contract which is resulting in generating a duplicate address which is already deployed on Sepolia and has more than 0 ETH balance.

Use your own Key to deploy the contract and hopefully your problem will be solved.

I could be wrong because I haven't seen your deploy scripts. If this does not solve your problem then please provide your project's Github repo so we can look for other possible problems in your code.

**Reply by dodeyeikoi** [reply]
Thanks for the help, I resolved it another way and I'll share how I did when I have my PC with me so I'll know if that's good too

**Reply by vasilsariev** [reply]
What's your solution of the problem? @dodeyeikoi
